### ISSUE TITLE
Variance-scaled stochastic gradient descent

### ISSUE THREAD

**Issue by accosmin** [accosmin/nano#177]
* crazy idea: scale (minibatch-averaged) gradient with its variance: dx = avg(g) / (1 + var(g))
* use a simple decreasing learning rate schedule
